src/com/google/javascript/jscomp/ConvertIIFEArgsToVars.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import com.google.common.base.Preconditions;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Convert the arguments of an immediately invoked function expression to variables
+ * and assign the value from the call arguments. This improves function inlining.
+ *
+ * Since passes which implement OptimizeCalls only handle named references, this is it's own pass.
+ *
+ * @author chadkillingsworth@gmail.com (Chad Killingsworth)
+ */
+public class ConvertIIFEArgsToVars extends NodeTraversal.AbstractPreOrderCallback
+    implements CompilerPass {
+  private final AbstractCompiler compiler;
+  private final List<Node> validIifes;
+
+  public ConvertIIFEArgsToVars(AbstractCompiler compiler) {
+    this.compiler = compiler;
+    validIifes = new ArrayList();
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+   NodeTraversal.traverseEs6(compiler, root, this);
+  }
+
+  public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
+    return true;
+  }
+
+  public void visit(NodeTraversal t, Node n, Node parent) {
```

Comment:
This doesn't check that the IIFE is an expr-result. Does that matter, or do you want to do this optimization for all iifes?